test/e2e/multikueue/e2e_test.go

```diff
@@ -270,6 +273,84 @@ var _ = ginkgo.Describe("MultiKueue", func() {
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			pods := &corev1.PodList{}
+			gomega.Expect(k8sManagerClient.List(ctx, pods, client.InNamespace(managerNs.Namespace),
+				client.MatchingLabels(deployment.Spec.Selector.MatchLabels))).To(gomega.Succeed())
+
+			// Check all worker pods are in a running phase.
+			ensurePodWorkloadsRunning(pods, *managerNs, multiKueueAc)
+
+			deploymentConditions := []appsv1.DeploymentCondition{
+				{
+					Type:    appsv1.DeploymentAvailable,
+					Status:  corev1.ConditionTrue,
+					Reason:  "MinimumReplicasAvailable",
+					Message: "Deployment has minimum availability.",
+				},
+				{
+					Type:   appsv1.DeploymentProgressing,
+					Status: corev1.ConditionTrue,
+					Reason: "NewReplicaSetAvailable",
+				},
+			}
+
+			ginkgo.By("Waiting for the deployment to get status updates", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdDeployment := appsv1.Deployment{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(deployment), &createdDeployment)).To(gomega.Succeed())
+					g.Expect(createdDeployment.Status.ReadyReplicas).To(gomega.Equal(int32(3)))
+					g.Expect(createdDeployment.Status.Replicas).To(gomega.Equal(int32(3)))
+					g.Expect(createdDeployment.Status.UpdatedReplicas).To(gomega.Equal(int32(3)))
+					g.Expect(createdDeployment.Status.Conditions).To(gomega.BeComparableTo(deploymentConditions, cmpopts.IgnoreFields(appsv1.DeploymentCondition{}, "LastTransitionTime", "LastUpdateTime", "Message"))) // Ignoring message as it is required to gather the pod replica set
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Increasing the replica count on the deployment", func() {
+				createdDeployment := &appsv1.Deployment{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).To(gomega.Succeed())
+					createdDeployment.Spec.Replicas = ptr.To[int32](4)
+					g.Expect(k8sManagerClient.Update(ctx, createdDeployment)).To(gomega.Succeed())
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+
+				ginkgo.By("Wait for replicas ready", func() {
+					createdDeployment := &appsv1.Deployment{}
+					gomega.Eventually(func(g gomega.Gomega) {
+						g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).To(gomega.Succeed())
+						g.Expect(createdDeployment.Status.ReadyReplicas).To(gomega.Equal(int32(4)))
+					}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+				})
+			})
+			pods = &corev1.PodList{}
+			gomega.Expect(k8sManagerClient.List(ctx, pods, client.InNamespace(managerNs.Namespace),
+				client.MatchingLabels(deployment.Spec.Selector.MatchLabels))).To(gomega.Succeed())
+
+			// Check all worker pods are in a running phase.
+			ensurePodWorkloadsRunning(pods, *managerNs, multiKueueAc)
```

Comment:
nit, actually instead of the comment could you wrap this in a `ginkgo.By("Check all worker pods are in a running phase")`

Reply:
Ah, I see, but still something feels off with the code being split outside and inside the helper function.

I would like to consider the following, move this code inside the function:
```
	pods = &corev1.PodList{}
	gomega.Expect(k8sManagerClient.List(ctx, pods, client.InNamespace(managerNs.Namespace),
		client.MatchingLabels(deployment.Spec.Selector.MatchLabels))).To(gomega.Succeed())
``` 
but keep the gingo.By outside. WDYT?

Reply:
(or make the ginkgo.By as the first line of the helper function)

Reply:
I like the first option better will make things more uniform, thanks mimowo